### PR TITLE
Improvements to the service cyborg experience

### PR DESCRIPTION
**Pull request by brndd**
## What this does

Tries to make service cyborgs less miserable to play by giving them a few new toys and tweaks.

### Drink synthesizer

2023-11-01T05:37:45

Adds a new drink synthesizer module to service cyborgs. It works similar to the material synthesizer: it has a list of chemicals it can create, and can scan new chemicals (based on a whitelist) from existing reagent containers (always picks the reagent with the most volume) and then create them later. It uses power from the cyborg's battery to synthesize chemicals, at a rate of 15 power per 1 unit (no modifiers for rarer chemicals due to technical limitations). The only things it can scan and synthesize are booze, soft drinks and hot drinks. Basically everything in the bar dispensers and booze-o-mat.

Clicking on reagent containers synthesizes the chemical directly into the container. It does not work on machines (e.g. the circuit imprinter), nor can it be used to directly force-feed people. Alt-clicking on it allows you to set the transfer amount, similar to reagent containers. Using it in hand toggles the mode between synthesizing and scanning. When the mode is toggled to synthesis, it asks you what chemical you want to synthesize. This is a little bit clumsy, but I couldn't figure out a better control scheme.

There is no way to get these in-game without adminbus, but there is also a human-usable version that by default can scan all the chemicals found in the ChemMaster. This version uses compressed matter cartridges as fuel. There is also an admin version that can scan and synthesize *anything* (admins should varedit the white- or blacklists before using it) and uses no compressed matter or power.

<details>
<summary>Usage example</summary>

2023-11-01T04:58:11
2023-11-01T04:58:35
2023-11-01T04:58:45

</details>

### Other changes

- Brobot's Space Beer and Mickey Finn's Special Brew (the chloral hydrate fake beer) have been removed because they are redundant with the addition of the drink synthesizer.
- Service cyborgs get a bartender's shaker.
- Service cyborgs get a rag for cleaning up messes. This rag can't be used to smother people... except when the borg is emagged.
- Service cyborgs now start with cooking equipment out of the box. The cooking upgrade board has been removed.
- Emagging a service borg now gives them a ~~meat cleaver (a chef traitor item). This could be a bit too robust so I don't mind downgrading it to the significantly less dangerous~~ butcher knife ~~or something, though contrary to its uplink description it does *not* deal similar damage to an esword (25 vs. 30, and 50 vs. 100 armor pen), and standard cyborgs get an esword, so..~~.
- Emagging a service borg unlocks a few new chemicals in the drink synthesizer:
  - Chloral hydrate disguised as beer (Mickey Finn's Special Brew).
  - Potassium, which can be mixed with water to explode things.
  - Ammonia and bleach, which can be mixed to create a smoke cloud of anti-lung gas (chloramine fumes).
- The Rapid Service Fabricator can synthesize directly into the service borg's gripper for extra elegancy. Sadly I couldn't enable passing items directly to other characters' hands from the gripper, because givecode is completely carbon-centric and I didn't want to refactor it that much.
- The Rapid Service Fabricator no longer creates sparks when selecting a design. Instead it just clicks. This applies to the standard handheld one too. Throwing sparks all over the bar is very inelegant and unfashionable.
- Updated the descriptions on a couple of service borg items to make them a little bit more descriptive.

## Why it's good

Service cyborgs are incredibly clunky even by cyborg standards and have very limited uses, making them miserable to play. Almost nobody ever plays them. This should make them slightly less miserable to play.

Other cyborg modules were far more versatile, so giving service cyborgs a bit more versatility by making cooking tools part of their base kit makes sense to me. The cooking tools being an upgrade board is like if medical borgs needed an upgrade board to do surgery. Hydroponics is still behind an upgrade board, because otherwise these guys wouldn't have any real upgrade boards.

I may have gone overboard with the toys they get when emagged, so feedback on that front is welcome.

## Changelog

:cl:
 * rscadd: Service cyborgs now have a general-purpose drink synthesizer to replace their old space beer. This device can scan most types of drinks and then synthesize them out of battery power. They also come with a shaker, a rag and cooking tools out of the box, and get spicy chemicals and a new toy when emagged.
 * rscdel: The service cyborg chef upgrade has been removed as it is now part of their standard loadout.
 * tweak: Service gripper can now pick up more things (basically everything the Rapid Service Fabricator can create, plus things needed in the kitchen). The gripper can now be directly synthesized to from the RSF.
 * tweak: The Rapid Service Fabricator no longer throws sparks all over the place when selecting what to fabricate. Instead it plays a clicking sound.

### CODE REVIEW

**Comment by pedr69** [comment]
i should really get around to finishing that serviceborg tackling module

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 1]
```diff
@@ -458,12 +458,28 @@
...
-	emag = new /obj/item/weapon/reagent_containers/glass/replenishing/cyborg/hacked(src)
+	emag = new /obj/item/weapon/kitchen/utensil/knife/large/butch/meatcleaver(src)
 
 	fix_modules()
 
+/obj/item/weapon/robot_module/butler/on_emag()
+	//add some spicy chemicals to the synth
+	for(var/M in modules)
+		if(istype(M, /obj/item/device/chem_synth/robot/service))
+			to_chat(world, "found")
```
Comment (pedr69) [inline code/modules/mob/living/silicon/robot/robot_modules.dm:476]: might wanna message_admins() instead or just remove this

Comment (brndd) [reply code/modules/mob/living/silicon/robot/robot_modules.dm]: Oh yeah, that's a debug line that I forgot in. Thanks.

**Comment by Dacendeth** [comment]
I dont like them getting traitor cleaver on being emagged in addition to their emagged chem unlocks.

**Comment by brndd** [comment]
On second thought me neither, so I've replaced it with the regular butcher cleaver that's a 5 damage upgrade over the kitchen knife (and a massive downgrade over the traitor one).

I still want them to have some moderately decent melee weapon because I think it's funny and fitting for the module to be more robust in combat than one might expect.

**Pull request merged**